File: src/main/java/superstitio/cards/general/SkillCard/gainEnergy/AddSexToy.java
package superstitio.cards.general.SkillCard.gainEnergy;

import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import superstitio.DataManager;
import superstitio.cards.general.GeneralCard;
import superstitio.cards.general.TempCard.SexToy;
import superstitio.utils.ActionUtility;

public class AddSexToy extends GeneralCard {
    public static final String ID = DataManager.MakeTextID(AddSexToy.class);

    public static final CardType CARD_TYPE = CardType.SKILL;

    public static final CardRarity CARD_RARITY = CardRarity.UNCOMMON;

    public static final CardTarget CARD_TARGET = CardTarget.SELF;

    private static final int COST = 1;
    private static final int MAGIC = 1;
//    private static final int UPGRADE_MAGIC = 1;

    public AddSexToy() {
        super(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET);
        this.setupMagicNumber(MAGIC);
        this.exhaust = true;
        this.cardsToPreview = new SexToy();
    }

    @Override
    public void use(AbstractPlayer player, AbstractMonster monster) {
        ActionUtility.addToBot_makeTempCardInBattle(new SexToy(), BattleCardPlace.Hand, this.magicNumber, this.upgraded);
    }

    @Override
    public void upgradeAuto() {
//        upgradeMagicNumber(UPGRADE_MAGIC);
        upgradeCardsToPreview();
    }
}

File: superstitioMain/java/superstitio/cards/general/SkillCard/gainEnergy/AddSexToy.kt
package superstitio.cards.general.SkillCard.gainEnergy

import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.monsters.AbstractMonster
import superstitio.DataManager
import superstitio.cards.general.GeneralCard
import superstitio.cards.general.TempCard.SexToy
import superstitioapi.utils.ActionUtility

class AddSexToy : GeneralCard(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET)
{
    //    private static final int UPGRADE_MAGIC = 1;
    init
    {
        this.setupMagicNumber(MAGIC)
        this.exhaust = true
        this.cardsToPreview = SexToy()
    }

    override fun use(player: AbstractPlayer?, monster: AbstractMonster?)
    {
        ActionUtility.addToBot_makeTempCardInBattle(
            SexToy(),
            ActionUtility.BattleCardPlace.Hand,
            this.magicNumber,
            this.upgraded
        )
    }

    override fun upgradeAuto()
    {
//        upgradeMagicNumber(UPGRADE_MAGIC);
        upgradeCardsToPreview()
    }

    companion object
    {
        val ID: String = DataManager.MakeTextID(AddSexToy::class.java)

        val CARD_TYPE: CardType = CardType.SKILL

        val CARD_RARITY: CardRarity = CardRarity.UNCOMMON

        val CARD_TARGET: CardTarget = CardTarget.SELF

        private const val COST = 2
        private const val MAGIC = 2
    }
}
